src/tests/dictionaryConcepts/components/CreateMapping.test.js

```diff
@@ -86,4 +87,183 @@ describe('Test suite for dictionary concepts components', () => {
     wrapper.setProps(nextProps);
     expect(wrapper.instance().props.map_type).toEqual('Associated With');
   });
+
+  it('should handle update of new mappings row source data when isNew is true', () => {
+    const newProps = {
+      ...props,
+      isNew: true,
+    };
+
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+    const inputField = wrapper.find('CreateMapping');
+    inputField.find('#source').simulate('change');
+    expect(props.updateEventListener).toHaveBeenCalled();
+  });
+
+  it('should handel update of new mappings row data for non Ciel concepts when isNew is true', () => {
+    const newProps = {
+      ...props,
+      isNew: true,
+      source: 'Classes',
+    };
+
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+    const inputField = wrapper.find('CreateMapping');
+    inputField.find('#to_concept_code').simulate('change');
+    expect(props.updateEventListener).toHaveBeenCalled();
+  });
+
+  it('should handle update of new mappings row concept name datew for Ciel concepts when isNew is true', () => {
+    const newProps = {
+      ...props,
+      isNew: true,
+      source: 'Ciel',
+    };
+
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+    const inputField = wrapper.find('CreateMapping');
+    inputField.find('#ConceptName').simulate('change');
+    expect(props.updateEventListener).toHaveBeenCalled();
+  });
+
+  it('should handle change when a user inputs concept name mappings data for for existing concepts', () => {
+    const newProps = {
+      ...props,
+      source: 'CIEL',
+    };
+
+    wrapper = mount(<Router>
+      <table><tbody><CreateMapping {...newProps} /></tbody></table>
+    </Router>);
+
+    const inputField = wrapper.find('CreateMapping');
+    const spy = jest.spyOn(inputField.instance(), 'handleInputChange');
+    inputField.find('#searchInputCiel').simulate('change');
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should handle change when a user inputs concept name mappings data for new rows when isNew is true', () => {
+    const newProps = {
+      ...props,
+      source: 'CIEL',
```

Comment:
Don't we have a constant for this?

Reply:
I have updated it to use the default source constant `INTERNAL_MAPPING_DEFAULT_SOURCE`